packages/machine-extractor/src/extractAction.ts

```diff
@@ -351,23 +268,16 @@ export const getActionCreatorName = (actionNode: ActionNode) =>
     ? actionNode.node.callee.name
     : undefined;
 
+// TODO: enforce return type to ActionEvent so `type` property is required
```

Comment:
what should happen if somebody writes this?
```ts
raise({ notAType: 'foo' })
```

Reply:
We still need to think about proper handling of invalid inputs but for now, maybe we can trust there's a type property always in the type level?